cmd/samples/pso/workflow.go

```diff
@@ -58,16 +58,16 @@ func PSOWorkflow(ctx workflow.Context, functionName string) (err error) {
 	// Set child workflow options
 	execution := workflow.GetInfo(ctx).WorkflowExecution
 	// Parent workflow can choose to specify it's own ID for child execution.  Make sure they are unique for each execution.
-	childID := fmt.Sprintf("PSO_child_workflow:%v", execution.RunID)
+	childWorkflowID := fmt.Sprintf("PSO_child_workflow:%v", execution.RunID)
 	cwo := workflow.ChildWorkflowOptions{
 		// Do not specify WorkflowID if you want cadence to generate a unique ID for child execution
-		WorkflowID:                   childID,
+		WorkflowID:                   childWorkflowID,
 		ExecutionStartToCloseTimeout: time.Minute,
 	}
 	ctx = workflow.WithChildOptions(ctx, cwo)
 
-	// Setup query handler for query type "stage"
-	ctx = workflow.WithValue(ctx, QueryResultName, childID)
+	// Setup query handler for query type "child"
+	ctx = workflow.WithValue(ctx, QueryResultName, childWorkflowID) // Don't need the runID when querying. Use only workflowID and it is going to query the latest one
 	err = workflow.SetQueryHandler(ctx, "child", func(input []byte) (string, error) {
 		return ctx.Value(QueryResultName).(string), nil
```

Comment:
Any reason that context value is used instead of just returning childWorklfowID variable?

Reply:
I was just trying to show how to create a global variable that is context specific. But yeah, not strictly necessary in this case